thrust/functional.h

```diff
@@ -139,6 +139,39 @@ struct binary_function
  *  \{
  */
 
+#define THRUST_UNARY_FUNCTOR_VOID_SPECIALIZATION(func, impl)                   \
+  template <>                                                                  \
+  struct func<void>                                                            \
+  {                                                                            \
+    __thrust_exec_check_disable__                                              \
+    template <typename T>                                                      \
+    __host__ __device__                                                        \
+    constexpr auto operator()(T&& x) const                                     \
+      noexcept(noexcept(impl)) -> decltype(impl)                               \
+    {                                                                          \
+      return impl;                                                             \
+    }                                                                          \
```

Comment:
Need to add `is_transparent` here, which is a type alias for a valid unspecified type (http://eel.is/c++draft/arithmetic.operations.plus).